presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java

```diff
@@ -97,6 +97,17 @@ public void testLimit()
         assertQuery("SELECT count(*) FROM (SELECT * FROM test_catalog.information_schema.tables LIMIT 1000)", "VALUES 1000");
     }
 
+    @Test
+    public void testRoleAuthorizationDescriptor()
+    {
+        assertQuery("SELECT count(*) FROM test_catalog.information_schema.role_authorization_descriptors", "VALUES 100");
+        assertQuery("SELECT count(*) FROM test_catalog.information_schema.roles", "VALUES 50");
+        assertQuery("SELECT count(*) FROM test_catalog.information_schema.enabled_roles", "VALUES 50");
+        assertQuery("SELECT count(*) FROM test_catalog.information_schema.applicable_roles", "VALUES 1");
+        assertQuery("SELECT role_name FROM test_catalog.information_schema.role_authorization_descriptors WHERE grantee = 'user5'", "VALUES ('role2')");
+        assertQuery("SELECT grantee FROM test_catalog.information_schema.role_authorization_descriptors WHERE role_name = 'role2'", "VALUES ('user4'), ('user5')");
```

Comment:
can you use `assertMetadataCalls` to make sure that predicate pushdown and limit works.

Reply:
Hmm... I'd have to look inside the function calls then (i.e. I'd have to know not only that a function was called, but also what was passed). Let me think about how I'd change the CountingMockConnector to do so.